app/src/main/java/de/westnordost/streetcomplete/map/MainFragment.kt

```diff
@@ -819,6 +831,29 @@ class MainFragment : Fragment(R.layout.fragment_main),
         }
     }
 
+    private fun showHighlightedElements(quest: OsmQuest, element: Element) {
+        val questType = quest.osmElementQuestType
+        val bbox = quest.geometry.center.enclosingBoundingBox(questType.highlightedElementsRadius)
+        var mapData: MapDataWithGeometry? = null
+
+        fun getMapData(): MapDataWithGeometry {
+            val data = mapDataWithEditsSource.getMapDataWithGeometry(bbox)
+            mapData = data
+            return data
+        }
+
+        viewLifecycleScope.launch {
+            val elements = withContext(Dispatchers.IO) { questType.getHighlightedElements(element, ::getMapData) }
+            for (e in elements) {
+                // don't highlight "this" element
+                if (element == e) continue
+                val geometry = mapData?.getGeometry(e.type, e.id) ?: continue
+                val icon = getPinIcon(e.tags)
+                putMarkerForCurrentQuest(geometry, icon, e.tags["name"] ?: e.tags["brand"])
```

Comment:
Maybe use the `getNameOrBrandOrOperatorOrRef` function from `QuestUtil.kt`?

Reply:
Where would that be an improvement? And would it potentially be confusing for some elements?

Reply:
I'm not sure where that title is displayed at all :sweat_smile:

Reply:
> I'm not sure where that title is displayed at all sweat_smile

Presumably on the shops as in the first example video?

> Where would that be an improvement?

Postboxes for example, would get no name from the current code, but operator + ref from:
https://github.com/streetcomplete/StreetComplete/blob/0857679a146abaf474f051f8e8d12225d9a10ff3/app/src/main/java/de/westnordost/streetcomplete/quests/QuestUtil.kt#L68-L83

> And would it potentially be confusing for some elements?

Theoretically maybe, but then it gives just one place to fine-tune that logic, rather than having different behaviour in here compared to in a quest.

Reply:
> Postboxes for example, would get no name from the current code, but operator + ref from

Not sure if this would be an improvement though. Then every single post box in Germany potentially has something like "Deutsche Post AG Huberstraße Mitte" displayed. (The "refs" if, stated at all, are actually streets/addresses/address descriptions)

If there are two post boxes directly next to each other, one of "xyz mail" and the other of "royal mail" or whatever, it should already be clear from the location which is which (which is right of which)

Reply:
> Not sure if this would be an improvement though. Then every single post box in Germany potentially has something like "Deutsche Post AG Huberstraße Mitte" displayed. (The "refs" if, stated at all, are actually streets/addresses/address descriptions)

Ah, in the UK they are fewer than 10 characters for the ref as a genuine reference number not a location.

What about if we add a boolean brief option to getNameOrBrandOrOperatorOrRef, for things like this, you'd set brief=true and it might only add the ref if it's less than 10 characters, and otherwise just show the operator, whereas for the quest itself, you clearly want it regardless so you'd set brief=false and it would always show it.

> If there are two post boxes directly next to each other, one of "xyz mail" and the other of "royal mail" or whatever, it should already be clear from the location which is which (which is right of which)

We don't get that specific issue in the UK as the postboxes are essentially all Royal Mail, but you'll often get two Royal Mail postboxes next to each other (with different refs). At least once, myself or someone else has got the refs offset and tagged against the wrong boxes, unpicking that in general is a real pain to do easily, but could be quickly spotted using this.

More generally in the UK I've seen a lot of stuff like benches and bins mapped on the wrong side of a path, or the wrong way round. If a post box has been removed, or two phone boxes have become one (i.e. one has gone), the more info referencing which one is which the better and the greater the chance of me making the right choice. I don't want to delete the left one if the remaining one is actually BT and the one I've deleted is the BT one (if that makes sense).

TL;DR, you should at least add operator to your chained selection list, at which point you may as well just use the existing function.

Reply:
I leave it as is for now